build.gradle.kts

```diff
@@ -21,50 +21,161 @@ buildscript {
     }
 }
 
+// Find property in either System environment or Gradle properties.
+// If set in both places, Gradle properties win.
+fun getPropertyValue(propertyName: String, throwIfNotFound: Boolean = false): String {
+    var value: String? = System.getenv(propertyName)
+    if (project.hasProperty(propertyName)) {
+        value = project.properties[propertyName] as String?
+    }
+    if (throwIfNotFound && (value == null || value.trim().isEmpty())) {
+        throw GradleException("Could not find '$propertyName'. " +
+                "Most be provided as either environment variable or " +
+                "a Gradle property.")
+    }
+    return value ?: ""
+}
+
+// Cache dir for build artifacts that should be stored on S3
+val releaseMetaDataDir = File("${buildDir}/outputs/s3")
+releaseMetaDataDir.mkdirs()
+
+fun readAndCacheVersion(): String {
```

Comment:
We could also consider to move our own version into top-level `version.txt` and read that from Config.kt. After all we know where to find that and does need the same level of IDE assistance to navigate, etc. as for all the other dependency version, etc. 🤷

Reply:
Yes, I have been thinking the same thing. But right now it is a bit annoying to do as the Config.kt file is built in isolation before anything else, so it is a bit difficult to make it more advanced than just listing constants.

But I do agree it will probably make some common edits easier. I'll create an issue tracking it.